src/__internal__/label/label.style.ts

```diff
@@ -56,7 +56,11 @@ export const StyledLabelContainer = styled.div<StyledLabelContainerProps>`
   align-items: center;
   margin-bottom: 8px;
 
-  ${({ align, inline, pr, pl, width }) =>
+  ${({ align }) => css`
+    justify-content: ${align === "right" ? "flex-end" : "flex-start"};
```

Comment:
question: @harpalsingh I notice when we set `flex-end` it flexes the whole width of the window/parent container, should we ensure it doesn't go past the width of the input or are you happy for us to push responsibility for that to implementing projects?

<img width="1556" alt="image" src="https://github.com/user-attachments/assets/fac5d0f7-78d8-47e7-ba69-fdf7b4914aad">

Reply:
Hey, for me it shouldn't go past the width of the input, otherwise the alignment would be off anyway? If it can be done inside the component update then it ensures people can accidentally make that error? If we push that responsibility, how would they handle that? if its simple enough then maybe provide instruction/demo in storybook for that.

Reply:
Consumers can control the width of the `StyledLabelContainer` with the `labelWidth` prop, so we could document its use to ensure the proper behaviour is achieved like so.

![Screenshot 2024-11-26 at 11 17 26](https://github.com/user-attachments/assets/0b9c4e53-99f2-499a-a5ef-baac0b914bfd)

However we would be left with an issue where if not explicity set we could see some incorrect behavioiur like in the screenshot posted by @edleeks87. I suggest we ensure the width of the label does not exceed that of the below inputs.

A pure css solution would be preferable 👍, but if not possible we may have to find a js solution.

Reply:
Thanks @harpalsingh, I think if we set `max-width` on the `FormField` we can fix the issue , but we'll need to do it conditionally based on the label props passed to the input components (there also may be a better way)
[example of changes](https://github.com/Sage/carbon/tree/demo-max-width-change)